### PR TITLE
Add typing for airflow/configuration.py

### PR DESCRIPTION
**Pull request by potiuk**
The configuraiton.py did not have typing information and it made
it rather difficult to reason about it-especially that it went
a few changes in the past that made it rather complex to
understand.

This PR adds typing information all over the configuration file

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @pingzh - prerequisite to understand and fix #23679

**Comment by potiuk** [comment]
Just to get a `little` more sanity on our configuration implementation.

**Comment by potiuk** [comment]
OK. I hope I have all static checks fixed now.

**Comment by potiuk** [comment]
All green.

**Review by pingzh — APPROVED** [review]
NICE. thanks

File: airflow/configuration.py [hunk 1]
```diff
@@ -620,15 +666,17 @@ def getjson(self, section, key, fallback=_UNSET, **kwargs) -> Union[dict, list,
         except (NoSectionError, NoOptionError):
             return default
 
-        if len(data) == 0:
+        if data is None or len(data) == 0:
```
Comment (turbaszek) [inline airflow/configuration.py]:
```suggestion
        if not data:
```
What do you think?

**Review by turbaszek — APPROVED** [review]
Small comment but otherwise looks good to me 🚀

File: airflow/configuration.py (hunk 1 above)
Comment (potiuk) [reply airflow/configuration.py]: Why not :)

**Comment by skyboi1233** [comment]
This was not me  I didn’t approve this it didn’t come from me


Sent from Yahoo Mail for iPhone


On Monday, May 16, 2022, 4:20 PM, Jarek Potiuk ***@***.***> wrote:




Merged #23716 into main.

**Pull request merged**